Catena4430_Sensor.ino

```diff
@@ -64,7 +64,14 @@ cTimer ledTimer;
 Catena::LoRaWAN gLoRaWAN;
 StatusLed gLed (Catena::PIN_STATUS_LED);
 
-cMeasurementLoop gMeasurementLoop;
+// the Temperature/Humidity Sensor
+cSHT3x gSht { Wire };
+
+// the light Sensor
+Ltr_329als gLtr {Wire};
```

Comment:
These need to be in a union, not at the top level, so that the cSHT3x and Ltr *objects* are not even available on the V1; and the BME / Si1133 objects are not available on the v2.  Here's how the data structures should be organized. 

```c++
class cMeasurementLoop {
  // things that are in any measurement loop variant, and then...

  virtual bool takeMeasurements() = 0;
  virtual bool formatMeasurements() = 0;
  virtual bool sendBuffer() = 0;
};

class cMeasurementLoopV1 : public cMeasurementLoop {
  // things specific to V1, including the measurement flags

  // and then
  cBME280 m_bme280; // constructor may need to initialize
  cSi1133 m_si1133;   // constructor may need to initialize

  virtual bool takeMeasurements() override;
  virtual bool formatMeasurements() override;
  virtual bool sendBuffer() override;
};

class cMeasurementLoopV2 : public cMeasurementLoop {
  // things specific to V2, including the measurement flags
  
  // constructor
  cMeasurementLoopV2( TwoWire_t &Wire)
       : m_sht: cSHT3x(Wire),
       , m_ltr: Ltr_329als(Wire)
       ;

  // and then
  cSHT3x m_sht:
  cLtr_329als m_ltr; 

  virtual bool takeMeasurements() override;
  virtual bool formatMeasurements() override;
  virtual bool sendBuffer() override;
};

union cMeasurementLoop {
  struct cMeasurementLoopV1 v1;
  struct cMeasurementLoopV2 v2;
};
```

There shall be 3 c++ source files: one with code for cMeasurementLoop, one with code for  cMeasrementLoopV1, and a third for cMeasurementLoopV2. There will correspondingly be 3 .h files; one for the base class, and one each for the sub-classes.

The code that implements `cMeasurementLoop` shall not refer to any globals for any of its local operations; it shall operate off the `this` pointer only, and will call the virtual methods when necessary to get more specific.

The top-level code that invokes cMeasurementLoop shall (at setup) generate a `cMeasurementLoop` pointer from the physical globals `gMeasurementLoopV1` and `gMeasurementLoop2`; this will happen in one place only, and otherwise those variables will not be directly referenced. This will be enforced by not making them visible except in a single file. In this way, we can delete the V1 code with zero risk or impact.